workspaces/mama/test/ManifestManager.spec.ts

```diff
@@ -0,0 +1,573 @@
+  ...kMinimalPackageJSON,
+  dependencies: {},
+  scripts: {},
+  license: "NONE"
+});
+
+describe("ManifestManager", () => {
+  describe("static Default", () => {
+    test("Property must be Frozen", () => {
+      const isUpdated = Reflect.set(ManifestManager.Default, "foo", "bar");
+
+      assert.strictEqual(isUpdated, false);
+      assert.ok(Object.isFrozen(ManifestManager.Default));
+    });
+  });
+
+  describe("static fromPackageJSON()", () => {
+    test(`Given a location equal to process.cwd(),
+      it should read and parse the JSON from filesystem`, async() => {
+      const readFile = sinon
+        .stub(fs, "readFile")
+        .resolves(JSON.stringify(kMinimalPackageJSON));
+
+      try {
+        const location = process.cwd();
+        const mama = await ManifestManager.fromPackageJSON(
+          location
+        );
+
+        assert.ok(readFile.calledOnce);
+        assert.ok(mama instanceof ManifestManager);
+
+        assert.strictEqual(
+          mama.isWorkspace,
+          false
+        );
+        assert.strictEqual(
+          mama.spec,
+          `${kMinimalPackageJSON.name}@${kMinimalPackageJSON.version}`
+        );
+      }
+      finally {
+        readFile.restore();
+      }
+    });
+
+    test(`Given a location equal to process.cwd() + package.json,
+      it should read and parse the JSON from filesystem`, async() => {
+      const readFile = sinon
+        .stub(fs, "readFile")
+        .resolves(JSON.stringify(kMinimalPackageJSON));
+
+      try {
+        const location = path.join(process.cwd(), "package.json");
+        const mama = await ManifestManager.fromPackageJSON(
+          location
+        );
+
+        assert.ok(readFile.calledOnce);
+        assert.ok(mama instanceof ManifestManager);
+
+        assert.strictEqual(
+          mama.isWorkspace,
+          false
+        );
+        assert.strictEqual(
+          mama.spec,
+          `${kMinimalPackageJSON.name}@${kMinimalPackageJSON.version}`
+        );
+      }
+      finally {
+        readFile.restore();
+      }
+    });
+
+    test("Given an invalid JSON, it should throw a custom Error with the parsing error as a cause", async(t) => {
+      const location = process.cwd();
+      const expectedLocation = path.join(location, "package.json");
+
+      const readFile = sinon
+        .stub(fs, "readFile")
+        .resolves(`{ foo: NaN }`);
+      t.after(() => readFile.restore());
+      // TODO: add t.plan(5) when available in LTS version of test_runner
+
+      try {
+        await ManifestManager.fromPackageJSON(
+          process.cwd()
+        );
+      }
+      catch (error) {
+        assert.strictEqual(error.name, "Error");
+        assert.strictEqual(
+          error.message,
+          `Failed to parse package.json located at: ${expectedLocation} (see the cause for additional information).`
+        );
+
+        assert.ok("cause" in error);
+        assert.strictEqual(error.cause.name, "SyntaxError");
+      }
```

Comment:
I guess its intended to not use `assert.rejects` here ?

Reply:
Yes, doesn't work well for cause